node/actors/network/src/testonly.rs

```diff
@@ -77,27 +77,32 @@ pub fn new_configs(
     setup: &validator::testonly::Setup,
     gossip_peers: usize,
 ) -> Vec<Config> {
-    let configs = setup.keys.iter().map(|key| {
-        let addr = net::tcp::testonly::reserve_listener();
-        Config {
-            server_addr: addr,
-            public_addr: (*addr).into(),
-            // Pings are disabled in tests by default to avoid dropping connections
-            // due to timeouts.
-            ping_timeout: None,
-            validator_key: Some(key.clone()),
-            gossip: GossipConfig {
-                key: rng.gen(),
-                dynamic_inbound_limit: usize::MAX,
-                static_inbound: HashSet::default(),
-                static_outbound: HashMap::default(),
-            },
-            max_block_size: usize::MAX,
-            tcp_accept_rate: limiter::Rate::INF,
-            rpc: RpcConfig::default(),
-            max_block_queue_size: 10,
-        }
-    });
+    let configs = setup
+        .validator_keys
+        .iter()
+        .zip(setup.attester_keys.iter())
```

Comment:
do not assume that setup contains the same number of validators and attesters

Reply:
Since the attesters are not yet used in the config, this change is stalled. However, I'll keep it in mind when implementing it.